app/logic/Calendar/Event.ts

```diff
@@ -83,15 +96,79 @@ export class Event extends Observable {
     this.duration = days * 86400;
   }
 
+  copyFromParent() {
+    let parentEvent = this.parentEvent;
+    this.calUID = parentEvent.calUID;
+    this.title = parentEvent.title;
+    this.descriptionText = parentEvent.descriptionText;
+    this.descriptionHTML = parentEvent.descriptionHTML;
+    this.allDay = parentEvent.allDay;
+    this.parentEvent = parentEvent;
+    this.location = parentEvent.location;
+    this.isOnline = parentEvent.isOnline;
+    this.isPresence = parentEvent.isPresence;
+    this.onlineMeetingURL = parentEvent.onlineMeetingURL;
+    this.participants.replaceAll(parentEvent.participants);
+  }
+
   async save() {
     assert(this.calendar, "To save an event, it needs to be in a calendar first");
     assert(this.calendar.storage, "To save an event, the calendar needs to be saved first");
     await this.calendar.storage.saveEvent(this);
+    for (let occurrence of this.instances) {
+      if (occurrence && !occurrence.dbID) {
+        occurrence.copyFromParent();
+      }
+    }
   }
 
   async deleteIt() {
     assert(this.calendar, "To delete an event, it needs to be in a calendar first");
     assert(this.calendar.storage, "To delete an event, the calendar needs to be saved first");
-    await this.calendar.storage.deleteEvent(this);
+    if (this.dbID) {
+      await this.calendar.storage.deleteEvent(this);
+    }
+    this.calendar.events.remove(this);
+    this.calendar.events.removeAll(this.instances);
+    if (this.parentEvent) {
+      let pos = this.parentEvent.instances.indexOf(this);
+      if (pos >= 0) {
+        this.parentEvent.instances[pos] = null;
```

Comment:
I specifically don't want to remove the element, I want to set it to null.

Reply:
Why? Who depends on that and why? Please make code comments to that effect.

Reply:
It's mainly for SQLEvent to know where the exclusions are so that it can save and restore them. The definition of `instances` does have a short comment, although I know you already want a longer comment for `recurrenceStartDate`.